### ISSUE TITLE
/* istanbul ignore if */ behaving weirdly in inline else/if chains

### ISSUE THREAD

**Issue by isaacs** [istanbuljs/istanbuljs#477]
I'm finding that an inline `/* istanbul ignore if */` does not behave as expected in cases like this:

```js
const test = n => {
  if (n === 1) {
    console.log(n)
  } else if (n === 2) {
    console.log(2)
  } else /* istanbul ignore if */ if (1 === 2) {
    console.log('math is broken')
  } else {
    console.log('n is some other thing')
  }
}

test(1)
test(2)
test(3)
// does not get 100% coverage
```

But this works fine:

```js
const test = n => {
  if (n === 1) {
    console.log(n)
  } else if (n === 2) {
    console.log(2)
  } else {
    /* istanbul ignore if */
    if (1 === 2) {
      console.log('math is broken')
    } else {
      console.log('n is some other thing')
    }
  }
}
test(1)
test(2)
test(3)
// gets 100% coverage, fails 100% coverage if test(3) omitted
```

I'd expect that the two code examples should logically have the same AST, so this is surprising.

Here's a gist, too: https://gist.github.com/isaacs/987fc367e2422ba782de9298586ea5c7

**Comment by coreyfarrell** [comment]
I recently found this issue, see #468.  A fix is merged to master and will be released in nyc@15.

The difference is that istanbul turns `else` into a block.  So:
```js
if (1) {
  console.log('1');
} else /* istanbul ignore if */ if (2) {
  console.log('2');
} else {
  console.log('other');
}
```

This becomes:
```js
if (1) {
  console.log('1');
} else /* istanbul ignore if */ {
  if (2) {
    console.log('2');
  } else {
    console.log('other');
  }
}
```

That is why `istanbul ignore next` works but it if/else ignores do not work.  The fix to #468 solves this by moving the comment into the `else` block so it is seen as `leadingComments` for the `if (2)`.

**Comment by isaacs** [comment]
Ah, perfect! In the meantime we'll just use an explicit block and put the comment inside.
